app/src/main/java/models/Post.java

```diff
@@ -10,7 +10,7 @@
 import java.util.Date;
 
 @ParseClassName("Post")
-public class Post extends ParseObject {
+public class Post extends ParseObject implements BasePost {
```

Comment:
nit: since you are implementing an interface, we generally add an @Override annotation to methods that are overridden. not critical but it helps us know quickly whether or not each method is self contained in this class, or possibly part of an interface/subclass